Make pint optional when running tests

<!-- Please fill in the following information to report a problem with Cantera. If you have a question about using Cantera, please post it on our Google Users' Group (https://groups.google.com/forum/#!forum/cantera-users). Feature enhancements should be discussed in the dedicated Cantera enhancements repository (https://github.com/Cantera/enhancements/new/choose) -->

**Problem description**

The use of the Pint units library within the Cantera Python module is optional. However, attempting to run the Python test suite without Pint installed fails.

This is dissimilar to how we handle testing of other optional dependencies in Python, where we are able to use the `@pytest.mark.skipif` decorator.

<!-- A clear and concise description of what the bug is. -->

**Steps to reproduce**

<!-- A minimal, complete, and reproducible example demonstrating the problem. See https://stackoverflow.com/help/minimal-reproducible-example for additional suggestions on how to create such an example. -->

1. Set up a Cantera build environment including all Python dependencies, but not including Pint
2. Run `scons test`

**Behavior**

```
___________________________________________ ERROR collecting test_units.py ___________________________________________
ImportError while importing test module '/root/cantera/test/python/test_units.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.9/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/python/test_units.py:7: in <module>
    import cantera.with_units as ctu
build/python/cantera/with_units/__init__.py:7: in <module>
    from pint import UnitRegistry, set_application_registry
E   ModuleNotFoundError: No module named 'pint'
```

**System information**

- Cantera version: 3.0.0b1
- OS: Debian 11 (Bullseye)
- Python/MATLAB/other software versions: Python 3.9

The bad part here is that the *entire Python test suite fails* as `pint` cannot be imported ... @bryanwweber do you have an angle on how to prevent this from happening? 

```
% scons test-python
[...]
=============================================== test session starts ===============================================
platform darwin -- Python 3.10.0, pytest-7.2.2, pluggy-1.2.0
rootdir: /Volumes/Data/work/GitHub/cantera/test/python, configfile: pytest.ini
plugins: anyio-3.7.1
collected 1867 items / 1 error

===================================================== ERRORS ======================================================
_________________________________________ ERROR collecting test_units.py __________________________________________
ImportError while importing test module '/Volumes/Data/work/GitHub/cantera/test/python/test_units.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/homebrew/Caskroom/miniforge/base/envs/cantera-dev/lib/python3.10/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
test/python/test_units.py:7: in <module>
    import cantera.with_units as ctu
build/python/cantera/with_units/__init__.py:7: in <module>
    from pint import UnitRegistry, set_application_registry
E   ImportError: cannot import name 'UnitRegistry' from 'pint' (unknown location)
------------------- generated xml file: /Volumes/Data/work/GitHub/cantera/test/work/pytest.xml --------------------
============================================= short test summary info =============================================
ERROR test/python/test_units.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
================================================ 1 error in 0.38s =================================================

```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -487,7 +487,7 @@ jobs:
         run: |
           conda install -q sundials=${{ matrix.sundials-ver }} scons numpy ruamel.yaml \
           cython boost-cpp fmt=${{ matrix.fmt-ver }} eigen yaml-cpp h5py pandas \
-          libgomp openblas pytest highfive pint
+          libgomp openblas pytest highfive
       - name: Build Cantera
         run: |
           scons build system_fmt=y system_eigen=y system_yamlcpp=y system_sundials=y \
@@ -621,7 +621,7 @@ jobs:
       - name: Install Python dependencies
         run: |
           python -m pip install -U pip setuptools wheel
-          python -m pip install '"scons<4.4.0"' pypiwin32 numpy ruamel.yaml cython pandas pytest pytest-github-actions-annotate-failures pint
+          python -m pip install '"scons<4.4.0"' pypiwin32 numpy ruamel.yaml cython pandas pytest pytest-github-actions-annotate-failures
       - name: Restore Boost cache
         uses: actions/cache@v3
         id: cache-boost

--- a/test/python/test_units.py
+++ b/test/python/test_units.py
@@ -4,7 +4,7 @@ from typing import Optional, Tuple, Dict
 import sys
 
 import pytest
-import cantera.with_units as ctu
+ctu = pytest.importorskip("cantera.with_units")
 import cantera as ct
 try:
     from pint.testing import assert_allclose

